### PR TITLE
fix(flagd): adjust flagd to updated error codes

### PR DESCRIPTION
**Pull request by leakonvalinka**
Fixing the inconsistencies regarding error codes for edge cases, as described in [#1679](https://github.com/open-feature/flagd/issues/1679)

### CODE REVIEW

File: providers/openfeature-provider-flagd/tests/e2e/inprocess/flags/allFlags.json [hunk 1]
```diff
@@ -0,0 +1,787 @@
+{
```
Comment (toddbaert) [inline providers/openfeature-provider-flagd/tests/e2e/inprocess/flags/allFlags.json:1]: @leakonvalinka @aepfli is this file supposed to be committed? This looks like the e2e flags we provide from the test-harness. I'm not sure why we'd need a copy here.

**Review by gruebel — APPROVED** [review]
looks good 🍻

File: providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/in_process.py [hunk 2]
```diff
@@ -128,30 +128,55 @@ def _resolve(
...
+    metadata: typing.Mapping[str, typing.Union[float, int, str, bool]],
+    reason: Reason,
+) -> FlagResolutionDetails:
+    variant, value = flag.default
+    if variant is None:
+        return FlagResolutionDetails(
+            value,
+            variant=variant,
+            reason=Reason.ERROR,
+            error_code=ErrorCode.FLAG_NOT_FOUND,
```
Comment (aepfli) [inline providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/in_process.py:175]: I compared this to the other methods, and there is a special error message used ""Flag '" + key+ "' has no default variant defined, will use code default"" maybe we should adapt the tests to add the message to be in sync in all of them? wdyt? @leakonvalinka
